File: test/Zorglub.Tests.CSharp/Time/Simple/Specialized/GregorianReformTests.cs
// SPDX-License-Identifier: BSD-3-Clause
// Copyright (c) 2020 Narvalo.Org. All rights reserved.

namespace Zorglub.Time.Simple.Specialized;

public static class GregorianReformTests
{
    [Fact]
    public static void Official_SecularShift() =>
        Assert.Equal(10, GregorianReform.Official.SecularShift);

    [Fact]
    public static void Official_FromLastJulianDate()
    {
        // Arrange
        var reform = GregorianReform.FromLastJulianDate(1582, 10, 4);
        // Assert
        Assert.Equal(reform, GregorianReform.Official);
    }

    [Fact]
    public static void Official_FromFirstGregorianDate()
    {
        // Arrange
        var reform = GregorianReform.FromFirstGregorianDate(1582, 10, 15);
        // Assert
        Assert.Equal(reform, GregorianReform.Official);
    }
}

File: test/Zorglub.Tests/Tests/Simple/Specialized/GregorianReformTests.fs
// SPDX-License-Identifier: BSD-3-Clause
// Copyright (c) 2020 Narvalo.Org. All rights reserved.

module Zorglub.Tests.Simple.Specialized.GregorianReformTests

open Zorglub.Testing

open Zorglub.Time.Simple.Specialized

open Xunit

[<Fact>]
let ``Property Official.SecularShift`` () =
    GregorianReform.Official.SecularShift === 10

[<Fact>]
let ``Property Official (from last Julian date)`` () =
    let reform = GregorianReform.FromLastJulianDate(1582, 10, 4)

    GregorianReform.Official === reform

[<Fact>]
let ``Property Official (from first Gregorian date)`` () =
    let reform = GregorianReform.FromFirstGregorianDate(1582, 10, 15)

    GregorianReform.Official === reform

